File: python/ circular_linked_list.py
class Node:

    def __init__(self, val):
        self.val = val
        self.next = None


class CircularLinkedList:

    def __init__(self):
        self.head = None

    def push(self, val):
        node = Node(val)

        node.next = self.head

        if self.head:
            temp = self.head
            while(temp.next != self.head):
                temp = temp.next
            temp.next = node
        else:
            node.next = node

        self.head = node

    def print_list(self):
        temp = self.head

        while(temp.next != self.head):
            print(temp.val)
            temp = temp.next

        print(temp.val)


if __name__ == '__main__':
    llist = CircularLinkedList()
    llist.push(6)
    llist.push(7)
    llist.push(1)
    llist.print_list()

File: go/circular_linked_list.go
package main

import (
	"fmt"
)

type Node struct {
	key  int
	next *Node
}

type CircularLinkedList struct {
	head *Node
}

func (circularLinkedList *CircularLinkedList) push(value int) {

	newNode := &Node{
		key: value,
	}

	if circularLinkedList.head == nil {
		circularLinkedList.head = newNode
		newNode.next = newNode
	}

	newNode.next = circularLinkedList.head

	curr := circularLinkedList.head

	for curr != nil {
		if curr.next == circularLinkedList.head {
			curr.next = newNode
			break
		}
		curr = curr.next
	}

	circularLinkedList.head = newNode

}

func (circularLinkedList *CircularLinkedList) print() {
	curr := circularLinkedList.head

	for curr.next != circularLinkedList.head {
		fmt.Println(curr.key)
		curr = curr.next
	}

	fmt.Println(curr.key)

	fmt.Println("-")
}

func main() {
	circularLinkedList := CircularLinkedList{}

	circularLinkedList.push(6)
	circularLinkedList.print()
	circularLinkedList.push(7)
	circularLinkedList.print()
	circularLinkedList.push(1)
	circularLinkedList.print()
}
